clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -114,8 +114,10 @@ public synchronized void add(String topic) {
      * is now
      */
     public synchronized long timeToNextUpdate(long nowMs) {
+        // Could be > 0 when update isn't requested explicitly and metadataExpiresMs isn't elapsed since the last successful update.
         long timeToExpire = needUpdate ? 0 : Math.max(this.lastSuccessfulRefreshMs + this.metadataExpireMs - nowMs, 0);
-        long timeToAllowUpdate = this.lastRefreshMs + this.refreshBackoffMs - nowMs;
+        // Could be > 0 when the last metadata update failed and refreshBackoffMs isn't elapsed yet since then.
```

Comment:
Same question as above, it seems that you mean "can be < 0" and it's not clear if we need a comment. In fact, I'd probably remove the Math.max in both of those lines.

In fact, I'd probably remove the `Math.max` from the two lines and simply have the last line as:

```java
Math.max(0, Math.max(timeToExpire, timeToAllowUpdate);
```